### ISSUE TITLE
Encoding Error when running bears on non-UTF-8 systems:

### ISSUE THREAD

**Issue by fneu** [coala/coala#2426]
Quoting @AbdealiJK here from https://github.com/coala-analyzer/docker-coala-base/pull/13

> This seems like a bug in coala. Essentially, the subprocess command must be converting the output from the program into the system's encoding. In the docker, the encoding is not UTF-8, while most of our systems must have utf-8. Which would cause a UnicodeDecodeError. We need to handle the decoding of the output properly.

Test output for LatexLintBear running on docker:

```
____________________________________________________________________________ LatexLintBearTest.test_invalid_files _____________________________________________________________________________

self = <tests.LocalBearTestHelper.verify_local_bear.<locals>.LocalBearTest testMethod=test_invalid_files>

    def test_invalid_files(self):
        self.assertIsInstance(invalid_files, (list, tuple))
        for file in invalid_files:
            self.check_validity(self.uut,
                                file,
                                filename,
                                valid=False,
                                force_linebreaks=force_linebreaks,
                                create_tempfile=create_tempfile,
>                               tempfile_kwargs=tempfile_kwargs)

tests/LocalBearTestHelper.py:203: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/LocalBearTestHelper.py:83: in check_validity
    execute_bear(local_bear, fname, file) as bear_output:
/usr/lib64/python3.5/contextlib.py:59: in __enter__
    return next(self.gen)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

bear = <LatexLintBear linter object (wrapping 'chktex') at 0x7f58d5215cf8>, args = ('/tmp/tmpf4qm2fpz', ['\n', '{ .}\n', '{ Sometext \\\n']), kwargs = {}, bear_output_generator = None
msg = ['Running bear LatexLintBear...', "Running 'chktex /tmp/tmpf4qm2fpz'", 'Bear LatexLintBear failed to run. Take a look ...decode(encoding)\nUnicodeDecodeError: 'utf-8' codec can't decode byte 0xde in position 116: invalid continuation byte"]

    @contextmanager
    def execute_bear(bear, *args, **kwargs):
        try:
            bear_output_generator = bear.execute(*args, **kwargs)
            assert bear_output_generator is not None, \
                "Bear returned None on execution\n"
            yield bear_output_generator
        except Exception as err:
            if isinstance(bear, Lint):  # Give extra debug info for Lint.
                if hasattr(bear, 'command'):
                    bear.debug("Command:", bear.command)
                if (hasattr(bear, 'stdout_output') and
                        isinstance(bear.stdout_output, collections.Iterable)):
                    bear.debug("Stdout:\n", "".join(bear.stdout_output))
                if (hasattr(bear, 'stderr_output') and
                        isinstance(bear.stderr_output, collections.Iterable)):
                    bear.debug("Stderr:\n", "".join(bear.stderr_output))
            msg = []
            while not bear.message_queue.empty():
                msg.append(bear.message_queue.get().message)
>           raise AssertionError(str(err) + " \n" + "\n".join(msg))
E           AssertionError: Bear returned None on execution
E            
E           Running bear LatexLintBear...
E           Running 'chktex /tmp/tmpf4qm2fpz'
E           Bear LatexLintBear failed to run. Take a look at debug messages (`-L DEBUG`) for further information.
E           The bear LatexLintBear raised an exception. If you are the writer of this bear, please make sure to catch all exceptions. If not and this error annoys you, you might want to get in contact with the writer of this bear.
E           
E           Traceback information is provided below:
E           
E           Traceback (most recent call last):
E             File "/coala/coalib/bears/Bear.py", line 231, in execute
E               return list(self.run_bear_from_section(args, kwargs) or [])
E             File "/coala/coalib/bears/Bear.py", line 224, in run_bear_from_section
E               return self.run(*args, **kwargs)
E             File "/coala/coalib/bearlib/abstractions/Linter.py", line 506, in run
E               stdin="".join(file) if options["use_stdin"] else None)
E             File "/coala/coalib/misc/Shell.py", line 96, in run_shell_command
E               ret = p.communicate(stdin)
E             File "/usr/lib64/python3.5/subprocess.py", line 1068, in communicate
E               stdout, stderr = self._communicate(input, endtime, timeout)
E             File "/usr/lib64/python3.5/subprocess.py", line 1741, in _communicate
E               self.stdout.encoding)
E             File "/usr/lib64/python3.5/subprocess.py", line 979, in _translate_newlines
E               data = data.decode(encoding)
E           UnicodeDecodeError: 'utf-8' codec can't decode byte 0xde in position 116: invalid continuation byte

tests/LocalBearTestHelper.py:36: AssertionError
```

**Comment by Makman2** [comment]
does python rely by default on the BOM? then we should not pass an encoding at all if not given^^

**Comment by fneu** [comment]
actually this is probably a dulicate of https://github.com/coala/coala-bears/issues/317 and fixed upstream.

**Comment by Makman2** [comment]
Did you rerun the stuff?

**Comment by fneu** [comment]
Not yet because I don't have the latest version of chktex available atm. But having reproduced this same backtrace on my native system I'm fairly certain that it is not a problem with docker (or any other non-utf-8 system) but with chktex itself that spits out non-utf-8 characters. 

@Abdeali has reported that bug upstream to chktex in April and it got apparently fixed in chktex 1.7.6 this September.

I guess we can make the bear depend on that version of chktex, but it will be a fair while until it's available... Tumbleweed and Arch are on 1.7.4 right now.

**Comment by Makman2** [comment]
alright :+1:
